### PR TITLE
Changes on the network classes

### PR DESCRIPTION
**Pull request by priyatharsan**
Replaced CNN-> ConvNet, DNN -> DenseNet, and added and removed network functionalities.
We will workaround using PRs to avoid conflicts.

### CODE REVIEW

**Comment by rfratila** [comment]
@priyatharsan please take a look at my comments above and see if you can address them

File: vulcanai2/models/basenetwork.py [hunk 1]
```diff
@@ -116,6 +116,48 @@ def stopping_rule(self, value):
...
+                                        padding=layer_param[4],
+                                        activation=self._activation))
+            #conv_layers.append(net_util.get_activation_fn(self.hid_layers_activation))
+            # Don't include batch norm in the first layer
+            #if self.batch_norm and i != 0:
+            #    conv_layers.append(nn.BatchNorm2d(hid_layer[1]))
+        conv_network = nn.Sequential(*conv_layers)
+        return conv_network
+
+    def build_dense_network(self, dims):
```
[the remark this answers is not in the document]
Comment (nirtiac) [reply vulcanai2/models/basenetwork.py]: Yeah I completely agree, this shouldn't be in basenetwork at all.

[the remark this answers is not in the document]
Comment (rfratila) [reply vulcanai2/models/basenetwork.py]: yes. basenetwork should contain generic code structure and helper functions that can be reused for any sort of module, whether it is dense, conv, or recurrent (in the future). it should be specific to any kind of network module

[the remark this answers is not in the document]
Comment (priyatharsan) [reply vulcanai2/models/basenetwork.py]: Alright then I will move those to functions to the respective classes

**Comment by rfratila** [comment]
Also a general comment, lets try to document each function so we don't have to do it later.

File: vulcanai2/models/cnn.py [hunk 2]
```diff
@@ -38,82 +38,22 @@ def _create_network(self):
         # Build Convolution Network
         self.conv_network = self.build_conv_network(self.conv_hid_layers)
         # Build Dense Network
-        self.dense_network = self.build_dense_network(self.dense_hid_layers)
-        # Build Network's Tail
+        self.conv_out_dim = self.get_conv_output_size()
+        dims = [self.conv_out_dim] + self.dense_hid_layers
+        self.dense_network = self.build_dense_network(dims)
```
[the remark this answers is not in the document]
Comment (priyatharsan) [reply vulcanai2/models/cnn.py]: you mean adding some sort of `if` condition?

[the remark this answers is not in the document]
Comment (rfratila) [reply vulcanai2/models/cnn.py]: Yeah the way I handled it before (it can probably be improved upon) but when creating a network, you had the option to specify how many classes will be predicted and that would toggle on or off the functionality and specify the output dimension you would need for the dense layer

File: vulcanai2/models/basenetwork.py [hunk 3]
```diff
@@ -116,6 +116,48 @@ def stopping_rule(self, value):
...
+        Helper function to calculate the size of the flattened 
+        features after the last conv layer
+        """
+        with torch.no_grad():
+            x = torch.ones(1, *self.in_dim)
+            x = self.conv_network(x)
+            return x.numel()
+
+    def build_conv_network(self, conv_hid_layers):
+        conv_dim = len(conv_hid_layers[0][2])
```
[the remark this answers is not in the document]
Comment (priyatharsan) [reply vulcanai2/models/basenetwork.py]: This hasn't been standardized yet. As of now it gets the inputs from `config['units']` which is a list of 2 list.

```network_conv_config = {
    'mode': 'conv',
    'units': ([ [1, 16, (5, 5), 2, 0],
                [16, 32, (5, 5), 1, 0],
                [32, 64, (5, 5), 1, 0] ],
              [ 100 ]),
    'dropouts': [0.2],
}```

That is: units = list(list(conv_hid_layer_and_parameters), list(dense_hid_layers))

[the remark this answers is not in the document]
Comment (rfratila) [reply vulcanai2/models/basenetwork.py]: I think we may be able to break down that 2D list so it's more intuitive to use and more explicit so mistakes don't get make. We can take advantage of the dictionary state that the config is specified in to add more detail within the units to specify filter count, filter size, stride, dilation factor, etc

File: vulcanai2/models/basenetwork.py [hunk 4]
```diff
@@ -155,6 +197,22 @@ def get_all_layers(self):
                 for param in l.parameters():
                     self.input_dimensions= param.size(0)
 
+    def init_layers(self, layers):
+        '''
+        Initializes all of the layers 
+        '''
+        bias_init = 0.01
```
[the remark this answers is not in the document]
Comment (priyatharsan) [reply vulcanai2/models/basenetwork.py]: sure, this was just a dry run actually. Also,I am having a thought of having this done with in the `LayerUnits`, what do you say? This is for just testing out how it'd look like if we do them on all layers altogether within network scope.

File: vulcanai2/models/basenetwork.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (priyatharsan) [reply vulcanai2/models/basenetwork.py]: oh yeah that makes more sense

[the remark this answers is not in the document]
Comment (rfratila) [reply vulcanai2/models/basenetwork.py]: you also have more flexibility in error checking for the 'unit' key-value correspondent for each type of network module you can make

File: vulcanai2/models/basenetwork.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (rfratila) [reply vulcanai2/models/basenetwork.py]: yes give it a shot and we can take another look

**Comment by priyatharsan** [comment]
Can you guys help me update the this task list as well following the review. I will be checking off one-by -one soon in upcoming PRs

- [ ] Break down `config['units']` to two: one for `_conv_` and the other for `_dense_`
- [ ]  add more detail within the units to specify filter count, filter size, stride, dilation factor, etc
- [ ] Generalize `get output size`
- [ ] Move back `build_conv_network` and `build_dense_network` to respective network classes

**Comment by rfratila** [comment]
I noticed we have quite a few testings scripts in Jupiter notebooks. Lets try to keep them in their own `examples` folder for now to keep the repo clean

File: vulcanai2/models/cnn.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (priyatharsan) [reply vulcanai2/models/cnn.py]: Gotcha!
